assets/js/src/game.js

```diff
@@ -0,0 +1,47 @@
+function Game() {
+    var self = this;
+    self.ui = new UI();
+
+    function getParameter(name) {
+        return name + '=' + $('[data-' + name + ']').val();
+    }
+
+    function getNewGameUri() {
+        return "new?" + getParameter('size') + '&' + getParameter('vs');
+    }
+
+    function getIdParameters() {
+        return 'player-id=' + self.playerId + '&game-id=' + self.gameId;
+    }
+
+    function listenForStatusChange (json) {
+        self.status = json['status'];
+        if (self.status && self.status != 'waiting')
+            self.ui.update(json, move);
+        else {
+            var uri = 'status?' + getIdParameters();
+            $.getJSON(uri, listenForStatusChange)
```

Comment:
It would be best to add a timeout in here somewhere. The way it is it's polling as fast as it can which will
essentially ddos your server